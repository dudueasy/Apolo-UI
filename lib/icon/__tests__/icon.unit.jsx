import * as renderer from 'react-test-renderer';
import React from 'react';
import Icon from '../icon';
import {shallow} from 'enzyme';


describe('Icon Component', () => {
  it('should render properly', () => {
    const IconProps = {name: 'qq'};
    const IconSnapshot = renderer.create(<Icon {...IconProps}/>).toJSON();
    expect(IconSnapshot).toMatchSnapshot();
  });

  it('call onClick when clicked', () => {
    const onClick = jest.fn(() => {
      console.log('Icon is clicked');
    });
    const IconProps = {name: 'qq', onClick};

    const wrapper = shallow(<Icon {...IconProps}/>);

    wrapper.simulate('click');
    wrapper.simulate('click');

    expect(onClick).toHaveBeenCalledTimes(2);
  });
});
