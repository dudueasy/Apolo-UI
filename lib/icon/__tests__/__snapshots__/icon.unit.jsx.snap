// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon FormExample should render properly 1`] = `
<svg
  className="apoloUI-icon"
>
  <use
    xlinkHref="#qq"
  />
</svg>
`;
